server/server.go

```diff
@@ -177,6 +178,14 @@ func (s *server) Run(rootCtx context.Context) (err error) {
 
 		s.gRPCServer.Stop()
 		zap.L().Warn("closed gRPC server")
+
+		if s.network != nil {
```

Comment:
Question: why are we applying this only to the `case err = <-gwErrc` case? Shouldn't this be done in all cases?

Reply:
we were only applying only to the rootCtx closure (dure to signal handling). But you are right. Moving this to outside the select.